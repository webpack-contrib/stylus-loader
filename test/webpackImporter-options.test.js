const path = require("path");

const {
  compile,
  getCodeFromBundle,
  getCodeFromStylus,
  getCompiler,
  getErrors,
  getWarnings,
} = require("./helpers");

jest.setTimeout(10000);

describe("webpackImporter option", () => {
  it("should work when value is not specify", async () => {
    const testId = "./import-webpack.styl";
    const compiler = getCompiler(
      testId,
      {},
      {
        resolve: {
          modules: [path.join(__dirname, "fixtures", "web_modules")],
        },
      }
    );
    const stats = await compile(compiler);
    const codeFromBundle = getCodeFromBundle(stats, compiler);
    const codeFromStylus = await getCodeFromStylus(testId);

    expect(codeFromBundle.css).toBe(codeFromStylus.css);
    expect(codeFromBundle.css).toMatchSnapshot("css");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
    expect(getErrors(stats)).toMatchSnapshot("errors");
  });

  it('should work when value is "true"', async () => {
    const testId = "./import-webpack.styl";
    const compiler = getCompiler(
      testId,
      {
        webpackImporter: true,
      },
      {
        resolve: {
          modules: [path.join(__dirname, "fixtures", "web_modules")],
        },
      }
    );
    const stats = await compile(compiler);
    const codeFromBundle = getCodeFromBundle(stats, compiler);
    const codeFromStylus = await getCodeFromStylus(testId);

    expect(codeFromBundle.css).toBe(codeFromStylus.css);
    expect(codeFromBundle.css).toMatchSnapshot("css");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
    expect(getErrors(stats)).toMatchSnapshot("errors");
  });

  it('should work when value is "false"', async () => {
    const testId = "./shallow-paths.styl";
    const compiler = getCompiler(testId, {
      webpackImporter: false,
      stylusOptions: {
        paths: ["test/fixtures/paths"],
      },
    });
    const stats = await compile(compiler);
    const codeFromBundle = getCodeFromBundle(stats, compiler);
    const codeFromStylus = await getCodeFromStylus(testId, {
      stylusOptions: {
        paths: ["test/fixtures/paths"],
      },
    });

    expect(codeFromBundle.css).toBe(codeFromStylus.css);
    expect(codeFromBundle.css).toMatchSnapshot("css");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
    expect(getErrors(stats)).toMatchSnapshot("errors");
  });

  it('should throw an error on webpack import when value is "false"', async () => {
    const testId = "./import-webpack.styl";
    const compiler = getCompiler(
      testId,
      {
        webpackImporter: false,
      },
      {
        resolve: {
          modules: [path.join(__dirname, "fixtures", "web_modules")],
        },
      }
    );
    const stats = await compile(compiler);

    await expect(
      getCodeFromStylus(testId, {
        stylusOptions: { shouldUseWebpackImporter: false },
      })
    ).rejects.toThrow();
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
    expect(getErrors(stats)).toMatchSnapshot("errors");
  });
});
