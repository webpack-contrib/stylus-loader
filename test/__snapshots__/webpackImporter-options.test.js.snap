// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webpackImporter option should throw an error on webpack import when value is "false": errors 1`] = `
[
  "ModuleBuildError: Module build failed (from \`replaced original path\`):
/test/fixtures/import-webpack.styl:1:9",
]
`;

exports[`webpackImporter option should throw an error on webpack import when value is "false": warnings 1`] = `[]`;

exports[`webpackImporter option should work when value is "false": css 1`] = `
".other {
  font-family: serif;
}
"
`;

exports[`webpackImporter option should work when value is "false": errors 1`] = `[]`;

exports[`webpackImporter option should work when value is "false": warnings 1`] = `[]`;

exports[`webpackImporter option should work when value is "true": css 1`] = `
".imported-in-web-modules {
  font-size: 2em;
}
.imported-in-web-modules {
  font-size: 2em;
}
"
`;

exports[`webpackImporter option should work when value is "true": errors 1`] = `[]`;

exports[`webpackImporter option should work when value is "true": warnings 1`] = `[]`;

exports[`webpackImporter option should work when value is not specify: css 1`] = `
".imported-in-web-modules {
  font-size: 2em;
}
.imported-in-web-modules {
  font-size: 2em;
}
"
`;

exports[`webpackImporter option should work when value is not specify: errors 1`] = `[]`;

exports[`webpackImporter option should work when value is not specify: warnings 1`] = `[]`;
